### PR TITLE
PROV-91 Added tests for methods in providerManagementUtils

### PR DESCRIPTION
**Pull request by jpetersen6202**
Added test cases for getSupervisors:
Contribution for PROV-91. Added three test cases for getSupervisors;
Testing that the list returned is not empty after setting a
supervisor/supervisee relationship, Testing that the list returned is
empty on a provider with no supervisor, and testing that the list returned is of
proper length when a supervisor/supervisee relationship exists.

Added test cases for clearTimeComponent:
Contribution for PROV-91. Added three test cases for clearTimeComponint;
Testing the date returned when passed a day that has already started (
(hour, min, second) != (0, 0, 0) ), testing the date returned when
passed a date with a negative value in the date, and testing the date
returned when passed a day that has just started ( (hour, min, second)
== (0, 0, 0) ).

Added test cases for getSupervisees:
Contribution for PROV-91. Added three test cases for getSupervisees;
Testing that the list returned is not empty after setting a
supervisor/supervisee relationship, Testing that the list returned is
empty on a provider with no supervisee, and testing that the list returned is of
proper length when a supervisor/supervisee relationship exists.

### CODE REVIEW

File: api/src/test/java/org/openmrs/module/providermanagement/ProviderManagementUtilsTest.java [hunk 1]
```diff
@@ -31,7 +48,107 @@
...
+    	Assert.assertEquals(date, result);
+    }
 
+	@Test
+   	public void shouldSetupContext() {
+   		assertNotNull(Context.getService(ProviderManagementService.class));
+   	}
+    
+    @Test
+    public void checkWhenSupervisorListIsNotEmpty() throws PersonIsNotProviderException, InvalidSupervisorException, ProviderAlreadyAssignedToSupervisorException {
```
Comment (mogoodrich) [inline api/src/test/java/org/openmrs/module/providermanagement/ProviderManagementUtilsTest.java]: Shouldn't this method be called something like "shouldAssignProviderToSupervisor"?

File: api/src/test/java/org/openmrs/module/providermanagement/ProviderManagementUtilsTest.java [hunk 2]
```diff
@@ -31,7 +48,107 @@
...
+    	provider.setIdentifier("1000X");
+    	Person person = Context.getPersonService().getPerson(6);
+    	Person supervisor = Context.getPersonService().getPerson(2);
+    	provider.setPerson(person);
+    	providerManagementService.assignProviderToSupervisor(person, supervisor);
+    	Assert.assertFalse(ProviderManagementUtils.getSupervisors(provider).isEmpty());
+    }
+    
+    @Test
+    public void checkWhenSupervisorListIsEmpty() throws PersonIsNotProviderException {
```
Comment (mogoodrich) [inline api/src/test/java/org/openmrs/module/providermanagement/ProviderManagementUtilsTest.java]: This should be called something like "supervisorListShouldBeEmptyIfNoSupervisors"

File: api/src/test/java/org/openmrs/module/providermanagement/ProviderManagementUtilsTest.java [hunk 3]
```diff
@@ -31,7 +48,107 @@
...
+    
+    @Test
+    public void checkForTheAccurateNumberOfSupervisorsAdded() throws PersonIsNotProviderException, InvalidSupervisorException, ProviderAlreadyAssignedToSupervisorException, DateCannotBeInFutureException {
+    	Provider provider = new Provider();
+    	provider.setIdentifier("1000X");
+    	Person person = Context.getPersonService().getPerson(6);
+    	Person supervisor = Context.getPersonService().getPerson(2);
+    	provider.setPerson(person);
+    	providerManagementService.assignProviderToSupervisor(person, supervisor);
+    	Assert.assertEquals(1, ProviderManagementUtils.getSupervisors(provider).size());
```
Comment (mogoodrich) [inline api/src/test/java/org/openmrs/module/providermanagement/ProviderManagementUtilsTest.java]: I'd just combine this with the "checkWhenSupervisorListIsNotEmpty" into a new "shouldAssignProviderToSupervisor"

File: api/src/test/java/org/openmrs/module/providermanagement/ProviderManagementUtilsTest.java [hunk 4]
```diff
@@ -31,7 +48,107 @@
...
+    }
+    
+    @Test
+    public void checkWhenSuperviseeListIsNotEmpty() throws PersonIsNotProviderException, InvalidSupervisorException, ProviderAlreadyAssignedToSupervisorException {
```
Comment (mogoodrich) [inline api/src/test/java/org/openmrs/module/providermanagement/ProviderManagementUtilsTest.java]: I'd make the same changes to the "supervisee list" methods that I suggest above for the "supervisor" list.

**Comment by jpetersen6202** [comment]
Thanks again, I'll work on these changes!

**Pull request closed without merging**